modin/core/dataframe/algebra/binary.py

```diff
@@ -19,11 +19,98 @@
 from .operator import Operator
 
 
+def coerce_int_to_float64(dtype: np.dtype) -> np.dtype:
+    """
+    Coerce dtype to float64 if it is a variant of integer.
+
+    If dtype is integer, function returns float64 datatype.
+    If not, returns the datatype argument itself.
+
+    Parameters
+    ----------
+    dtype : np.dtype
+        NumPy datatype.
+
+    Returns
+    -------
+    dtype : np.dtype
+        Returns float64 for all int datatypes or returns the datatype itself
+        for other types.
+
+    Notes
+    -----
+    Used to precompute datatype in case of division in pandas.
+    """
+    if dtype in np.sctypes["int"] + np.sctypes["uint"]:
+        return np.dtype(np.float64)
+    else:
+        return dtype
+
+
+def compute_dtypes_common_cast(first, second) -> np.dtype:
+    """
+    Precompute data types for binary operations by finding common type between operands.
+
+    Parameters
+    ----------
+    first : PandasQueryCompiler
+        First operand for which the binary operation would be performed later.
+    second : PandasQueryCompiler
+        Second operand for which the binary operation would be performed later.
+
+    Returns
+    -------
+    dtypes
+        The pandas series with precomputed dtypes.
+
+    Notes
+    -----
+    The dtypes of the operands are supposed to be known.
+    """
+    dtypes_first = dict(zip(first.columns, first._modin_frame._dtypes))
+    dtypes_second = dict(zip(second.columns, second._modin_frame._dtypes))
+    columns_first = set(first.columns)
+    columns_second = set(second.columns)
+    common_columns = columns_first.intersection(columns_second)
+    mismatch_columns = columns_first.union(columns_second) - common_columns
+    # If at least one column doesn't match, the result of the non matching column would be nan.
+    nan_dtype = np.dtype(type(np.nan))
+    dtypes = pandas.Series(
+        [
+            pandas.core.dtypes.cast.find_common_type(
+                [
+                    dtypes_first[x],
+                    dtypes_second[x],
+                ]
+            )
+            for x in common_columns
+        ],
+        index=common_columns,
+    )
+    dtypes = pandas.concat(
+        [
+            dtypes,
+            pandas.Series(
+                [nan_dtype] * (len(mismatch_columns)),
+                index=mismatch_columns,
+            ),
+        ]
+    )
+    dtypes = dtypes.sort_index()
+    return dtypes
+
+
 class Binary(Operator):
     """Builder class for Binary operator."""
 
     @classmethod
-    def register(cls, func, join_type="outer", labels="replace"):
+    def register(
+        cls,
+        func,
+        join_type="outer",
+        labels="replace",
+        how_compute_dtypes=None,
```

Comment:
```suggestion
        infer_dtypes=None,
```
I suggest renaming this to `infer_dtypes`.

Reply:
renamed to infer_dtypes.